pyproject_metadata/__init__.py

```diff
@@ -257,6 +257,9 @@ def __setattr__(self, name: str, value: Any) -> None:
         # update dynamic when version is set
         if name == 'version' and hasattr(self, 'dynamic'):
             self._update_dynamic(value)
+        # Can't be multiline
+        if name == 'description' and value and '\n' in value:
+            raise ConfigurationError('The description must be a single line')
```

Comment:
Why is `description` validated here but all other fields are validated only as part of the `from_pyproject()` class method?

Reply:
Ahh, probably should put it there. But it does mean that the validation doesn't happen if you manually create or change it. But that can be worked on (for all fields) later.

Reply:
I was asking because I think it is better to keep consistency (all fields validated when set, or all field validated via the constructor only), but I admit that I don't know why the code is structured the way it is, thus there may have been a reason for breaking the consistency.